server/ServerFSM.cpp

```diff
@@ -2167,62 +2164,53 @@ sc::result WaitingForMPGameJoiners::react(const JoinGame& msg) {
             }
         }
 
-        // verify that there is room left for this player
-        int already_connected_players = m_expected_ai_player_names.size() + server.m_networking.NumEstablishedPlayers();
-        if (already_connected_players >= m_num_expected_players) {
-            // too many human players
-            ErrorLogger(FSM) << "WaitingForSPGameJoiners.JoinGame : A human player attempted to join the game but there was not enough room.  Terminating connection.";
-            server.m_networking.Disconnect(player_connection);
-        } else {
-
-            std::string original_player_name = player_name;
+        std::string original_player_name = player_name;
 
-            // Remove AI prefix to distinguish Human from AI.
-            std::string ai_prefix = UserString("AI_PLAYER") + "_";
-            while (player_name.compare(0, ai_prefix.size(), ai_prefix) == 0)
-                player_name.erase(0, ai_prefix.size());
-            if(player_name.empty())
-                player_name = "_";
+        // Remove AI prefix to distinguish Human from AI.
+        std::string ai_prefix = UserString("AI_PLAYER") + "_";
+        while (player_name.compare(0, ai_prefix.size(), ai_prefix) == 0)
+            player_name.erase(0, ai_prefix.size());
+        if(player_name.empty())
+            player_name = "_";
 
-            std::string new_player_name = player_name;
+        std::string new_player_name = player_name;
 
-            bool collision = true;
-            std::size_t t = 1;
-            while (collision &&
-                   t <= m_lobby_data->m_players.size() + server.Networking().GetCookiesSize() + 1)
-            {
-                collision = false;
-                roles.Clear();
-                if (!server.IsAvailableName(new_player_name) || server.IsAuthRequiredOrFillRoles(new_player_name, roles)) {
-                    collision = true;
-                } else {
-                    for (std::pair<int, PlayerSetupData>& plr : m_lobby_data->m_players) {
-                        if (plr.second.m_empire_name == new_player_name) {
-                            collision = true;
-                            break;
-                        }
+        bool collision = true;
+        std::size_t t = 1;
+        while (collision &&
```

Comment:
Maybe a comment here explaining what is going on with name collisions and authorization to resolve them.

Reply:
Added.